pkg/validation/limits.go

```diff
@@ -959,6 +970,27 @@ func (o *Overrides) StreamRetention(userID string) []StreamRetention {
 	return o.getOverridesForUser(userID).StreamRetention
 }
 
+// RetentionHours returns the retention period for a given user.
+func (o *Overrides) RetentionHours(userID string, ls labels.Labels) string {
```

Comment:
I think we should use this one instead which is the one the compactor actually uses
https://github.com/grafana/loki/blob/4c88be0ef2310406206c02d7747bf73b51c95bda/pkg/compactor/retention/expiration.go#L134